### ISSUE TITLE
[Bug?]: Headers is not defined at DbAuthHandler._badRequest

### ISSUE THREAD

**Issue by LotuxPunk** [redwoodjs/redwood#11712]
### What's not working?

I'm trying to get a deployment working again, it was working amazingly on 7.1.3 but 8.4.0 is wrong for some reason.

I followed the upgrade guide and compared the files with other projects that are running in prod with RW 8.4.0, tried to clean the node_module, yarn.lock + cache but no chance, it still give me this error in prod when I try to logIn.

Can't reproduce the error by doing `yarn rw build` + `yarn rw serve`, I'm kinda lost. Does anyone have a lead on this ?

Server is running NodeJS 20.18
![image](https://github.com/user-attachments/assets/6e72a6d8-da9e-44ea-95a7-03230337eb24)

```
8|staging- | api | {
  "level":50,
  "time":1730493049871,
  "pid":2076641,
  "hostname":"kameo",
  "reqId":"req-5",
  "err":{
    "type":"ReferenceError",
    "message":"Headers is not defined",
    "stack":"ReferenceError: Headers is not defined\n
        at DbAuthHandler._badRequest (/var/www/staging/20241101201418/node_modules/@redwoodjs/auth-dbauth-api/dist/DbAuthHandler.js:918:20)\n
        at DbAuthHandler.invoke (/var/www/staging/20241101201418/node_modules/@redwoodjs/auth-dbauth-api/dist/DbAuthHandler.js:215:16)\n
        at async __rw_handler (/var/www/staging/20241101201418/api/dist/functions/auth.js:194:10)\n
        at async requestHandler (/var/www/staging/20241101201418/node_modules/@redwoodjs/api-server/dist/requestHandlers/awsLambdaFastify.js:92:30)"
    },
    "msg":"Headers is not defined"
  }
}
```
In `DbAuthHandler.js`
```js
  _badRequest(message) {
    return {
      statusCode: 400,
      body: JSON.stringify({ error: message }),
      headers: new Headers({ "content-type": "application/json" }) // This seems to be the problem
    };
  }
```

### What's your environment? (If it applies)

```shell
System:
    OS: Linux 5.15 Ubuntu 22.04.4 LTS 22.04.4 LTS (Jammy Jellyfish)
    Shell: 5.1.16 - /bin/bash
  Binaries:
    Node: 20.18.0 - /tmp/xfs-b851efa7/node
    Yarn: 3.2.1 - /tmp/xfs-b851efa7/yarn
  npmPackages:
    @redwoodjs/auth-dbauth-setup: 8.4.0 => 8.4.0 
    @redwoodjs/cli-data-migrate: 8.4.0 => 8.4.0 
    @redwoodjs/core: 8.4.0 => 8.4.0 
    @redwoodjs/project-config: 8.4.0 => 8.4.0 
  redwood.toml:
    [web]
      title = "Kameo"
      port = "${WEB_PORT:8910}"
      apiUrl = "/.redwood/functions" # you can customise graphql and dbauth urls individually too: see https://redwoodjs.com/docs/app-configuration-redwood-toml#api-paths
      includeEnvironmentVariables = ["LYCHEE_URL"] # any ENV vars that should be available to the web side, see https://redwoodjs.com/docs/environment-variables#web
    [api]
      port = "${API_PORT:8911}"
    [browser]
      open = true
    [generate]
      tests = true
      stories = false
    [notifications]
      versionUpdates = ["latest"]
```

### Are you interested in working on this?

- [ ] I'm interested in working on this

**Comment by LotuxPunk** [comment]
Any leads on how to debug this ?

**Comment by Tobbe** [comment]
@LotuxPunk Sorry for the delay

How/where are you deploying? Is this a Baremetal deployment?

**Comment by LotuxPunk** [comment]
> @LotuxPunk Sorry for the delay
> 
> How/where are you deploying? Is this a Baremetal deployment?

No worries 👍 

Absolutely, this is a Baremetal deployment, you can find the pm2 setup and the environnement on the first message.

**Comment by Tobbe** [comment]
Is the env output from your local machine or from the server?

**Comment by LotuxPunk** [comment]
From the server directly, I ran it on the latest deployment IIRC or in the root folder (with all the kept deployment to be able to rollback)

Here's the deploy.toml value for this deployment:
 ```toml
 [[staging.servers]]
host = "**********"
username = "*********"
password = "**********"
agentForward = false
sides = ["api","web"]
packageManagerCommand = "yarn"
monitorCommand = "pm2"
path = "/var/www/staging"
processNames = ["staging-serve"]
repo = "git@github.com:LotuxPunk/KameoManager.git"
branch = "develop"
keepReleases = 5
freeSpaceRequired = 2048
```

**Comment by Tobbe** [comment]
I'm deploying to Baremetal as well, and I use dbAuth.

Here's from my server
```
  System:
    OS: Linux 6.8 Ubuntu 24.04.1 LTS 24.04.1 LTS (Noble Numbat)
    Shell: 5.2.21 - /bin/bash
  Binaries:
    Node: 20.17.0 - /usr/bin/node
    Yarn: 4.4.0 - /usr/bin/yarn
  Databases:
    SQLite: 3.45.1 - /usr/bin/sqlite3
  npmPackages:
    @redwoodjs/auth-dbauth-setup: 8.3.0 => 8.3.0
    @redwoodjs/cli-data-migrate: 8.3.0 => 8.3.0
    @redwoodjs/core: 8.3.0 => 8.3.0
    @redwoodjs/project-config: 8.3.0 => 8.3.0
  redwood.toml:
    [web]
      title = "Redwood App"
      port = 8910
      apiUrl = "/api"
      includeEnvironmentVariables = [
        # Add any ENV vars that should be available to the web side to this array
        # See https://redwoodjs.com/docs/environment-variables#web
      ]
    [api]
      port = 8911
    [generate]
      tests = false
      stories = false
    [browser]
      open = true
    [notifications]
      versionUpdates = ["latest"]
```

So, two things for you to try:
1. Downgrade to rw 8.3.0
2. Upgrade to yarn 4.4.0

**Comment by LotuxPunk** [comment]
I give it a go, I keep you posted

**Comment by LotuxPunk** [comment]
Sort of good news, I have a different error message


Here's the `yarn rw info`
```
  System:
    OS: Linux 5.15 Ubuntu 22.04.4 LTS 22.04.4 LTS (Jammy Jellyfish)
    Shell: 5.1.16 - /bin/bash
  Binaries:
    Node: 20.18.0 - /tmp/xfs-b0745342/node
    Yarn: 4.4.0 - /tmp/xfs-b0745342/yarn
  npmPackages:
    @redwoodjs/auth-dbauth-setup: 8.3.0 => 8.3.0 
    @redwoodjs/cli-data-migrate: 8.3.0 => 8.3.0 
    @redwoodjs/core: 8.3.0 => 8.3.0 
    @redwoodjs/project-config: 8.3.0 => 8.3.0 
  redwood.toml:
    [web]
      title = "Kameo"
      port = "${WEB_PORT:8910}"
      apiUrl = "/.redwood/functions" # you can customise graphql and dbauth urls individually too: see https://redwoodjs.com/docs/app-configuration-redwood-toml#api-paths
      includeEnvironmentVariables = ["LYCHEE_URL"] # any ENV vars that should be available to the web side, see https://redwoodjs.com/docs/environment-variables#web
    [api]
      port = "${API_PORT:8911}"
    [browser]
      open = true
    [generate]
      tests = true
      stories = false
    [notifications]
      versionUpdates = ["latest"]
```


-----

But every time I attempt to login on the deployment, I get a 400

`{"error":"Username va****@gmail.com not found"}`


```
14|api  | {"level":30,"time":1731166321400,"pid":2419973,"hostname":"kameo","reqId":"req-9","req":{"method":"GET","url":"/auth?method=getToken","hostname":"redwood_server","remoteAddress":"127.0.0.1","remotePort":59286},"msg":"incoming request"}
14|api  | {"level":30,"time":1731166321403,"pid":2419973,"hostname":"kameo","reqId":"req-9","res":{"statusCode":200},"responseTime":2.602508544921875,"msg":"request completed"}
```

Despite it exist on the database:

```
kameo_manager=# SELECT   id,  COALESCE(email, 'NULL') AS email FROM "User";
            id             |             email             
---------------------------+-------------------------------
 cl729******* | va******@gmail.com
```

**Comment by LotuxPunk** [comment]
Also, I am wondering, I running my staging RedwoodJS deployment on the very same server/baremetal than the prod's one

I set the `cookieName` in the staging deployment (the one we are talking about) could it be that there's some interference between the 2 instances ?

**Comment by LotuxPunk** [comment]
Ok, I managed to get it working.

It turns out 8.3.0 is working fine, and I assume 8.4.0 is as well. My nginx configuration was outdated (I started this project back when “Baremetal” was called Serverful, probably before version 1.0.0, so the nginx config was quite deprecated). I updated it, but since there were multiple instances of RedwoodJS running on the server, I messed up the upstream server in those configurations. All requests were forwarded to the wrong API instance, which didn’t have this user (in fact, it wasn’t an instance of the same project at all).

Thanks a lot @Tobbe ! 🙏
